src/Populator/ArrayPropertyMappingPopulator.php

```diff
@@ -18,52 +18,64 @@
  */
 final class ArrayPropertyMappingPopulator implements Populator
 {
-    /** @var \Closure(mixed, TContext=):mixed */
-    private \Closure $mapper;
+    private ?string $sourceArrayItemProperty;
+    /** @var \Closure(mixed, TContext=):mixed|null */
+    private ?\Closure $mapper;
     private PropertyAccessorInterface $arrayItemAccessor;
-    private PropertyAccessorInterface $accessor;
+    private PropertyMappingPopulator $populator;
 
     /**
      * @param \Closure(mixed, TContext=):mixed|null $mapper
      */
     public function __construct(
-        private string $targetProperty,
-        private string $sourceArrayProperty,
-        private ?string $sourceArrayItemProperty = null,
+        string $targetProperty,
+        string $sourceArrayProperty,
+        ?string $sourceArrayItemProperty = null,
```

Comment:
Warum müssen die Modi hier default statt private sein?

Reply:
Do you mean why no property promotion is used here? Because half of the parameters are not properties and I find it easier to read this way.